lib/dynflow/testing/managed_clock.rb

```diff
@@ -17,12 +17,16 @@ def ping(who, time, with_what = nil, where = :<<)
         @pending_pings.sort!
       end
 
-      def progress
+      def progress(ignored_subjects = [])
         if next_ping = @pending_pings.shift
-          # we are testing an isolated system = we can move in time
-          # without actually waiting
-          @current_time = next_ping.when
-          next_ping.apply
+          if next_ping.what.respond_to?(:value) && ignored_subjects.include?(next_ping.what.value)
+            progress(ignored_subjects)
```

Comment:
There is a risk now, that we would not ignore some ping that is not related to the specific task, and we could eventually get stuck as well. I think we should do just plain `return` here, and in `progress_all`, just do `progress(ignored_subjects) until @pending_pings.empty?`

I know I'm getting annoying here, but we're getting threre :)

Reply:
@iNecas updated